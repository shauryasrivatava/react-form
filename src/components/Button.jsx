import React from 'react';

const Button = (props) => {
    const Tag = props.href? 'a':'button';
    return <Tag {...props} />
}

export default Button;