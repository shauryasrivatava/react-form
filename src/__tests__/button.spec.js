import renderer from 'react-test-renderer';
import Button from '../components/Button';
import React from 'react';
import { shallow } from 'enzyme';
import enzymeConfig from '../../enzymeConfig'

describe('<Button />', function () {

    it('Should render correctly', function () {
        const component = renderer.create(<Button />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should have single <a> when href prop exists', function () {
        const component = shallow(<Button href="check.com"/>);
        expect(component.find('a').length).toBe(1);
    });

    it('Should have single <button> when href prop exists', function () {
        const component = shallow(<Button />);
        expect(component.find('button').length).toBe(1);
    });

})