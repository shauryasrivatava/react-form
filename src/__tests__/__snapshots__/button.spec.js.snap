// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render correctly 1`] = `<button />`;
